lib/config/options.rb

```diff
@@ -127,6 +127,7 @@ def merge!(hash)
       current = to_hash
       DeepMerge.deep_merge!(hash.dup,
                             current,
+                            merge_nil_values: true,
```

Comment:
Note, this is a bit controversial because it's changing behavior.  If needed, this could be optional, defaulted to false (to maintain current behavior) and you can enable it via `Config.merge_nil_values` similar to the overwrite_arrays option.

so, the question is:
1) should this be optional
2) if so, what should be the default

I can change the PR based on this feedback.  Note, this is to fix #13.  Whatever is decided, we can update that issue with the right way solve that issue.